provider-assignments/src/java/org/sakaiproject/archiver/provider/AssignmentsArchiver.java

```diff
@@ -181,11 +228,34 @@ private void archiveAttachments(final List<Reference> attachments, final String
 	 * @throws IdUnusedException
 	 * @throws TypeException
 	 */
-	private void archiveAttachment(final Reference attachment, final String archiveId, final String siteId, final String subdir)
+	private void archiveAttachment(final Reference attachment, final String archiveId, final String siteId, final String subdir,
+			final String finalFolder)
 			throws ServerOverloadException, PermissionException, IdUnusedException, TypeException {
 		final byte[] attachmentBytes = this.contentHostingService.getResource(attachment.getId()).getContent();
-		this.archiverService.archiveContent(archiveId, siteId, TOOL_NAME, attachmentBytes,
-				attachment.getProperties().getPropertyFormatted(attachment.getProperties().getNamePropDisplayName()), subdir);
+		final String attachmentName = attachment.getProperties().getPropertyFormatted(attachment.getProperties().getNamePropDisplayName());
+		this.archiverService.archiveContent(archiveId, siteId, TOOL_NAME, attachmentBytes, attachmentName, subdir + finalFolder);
+		addToAttachmentsHtml(finalFolder, attachmentName);
+	}
+
+	/**
+	 * Set the html string that contains a list of attachment hyperlinks
+	 *
+	 * @param attachmentLocation
+	 * @param attachmentName
+	 */
+	private void addToAttachmentsHtml(final String attachmentLocation, final String attachmentName) {
```

Comment:
why not just pass the list of attachments to this and build the list in one go? Then you dont need finaliseAttachmentsHtml